### PR TITLE
AWS: More support for ap-northeast-2 region

### PR DESCRIPTION
**Pull request by matthewrudy**
Issue #24446

The new AWS region for Seoul, Korea (ap-northeast-2)
was launched in January 2016
https://aws.amazon.com/blogs/aws/now-open-aws-asia-pacific-seoul-region/

But it requires a few changes.

To test:

```
export KUBERNETES_PROVIDER=aws
export KUBE_AWS_ZONE=ap-northeast-2a
export MASTER_SIZE=t2.medium
export NODE_SIZE=t2.medium
export NUM_NODES=4
cluster/kube-up.sh
```

I assigned the AMIs by checking the specific version used from `ap-northeast-1`,
and finding the same image with the same datestamp.

---

This change is [Reviewable](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/24464)

### CODE REVIEW

**Comment by matthewrudy** [comment]
ok to test

File: cluster/aws/util.sh [hunk 1]
```diff
@@ -360,6 +360,10 @@ function detect-trusty-image () {
         AWS_IMAGE=ami-93876e93
         ;;
 
+      ap-northeast-2)
+        AWS_IMAGE=ami-62ac620c
```
Comment (justinsb) [inline cluster/aws/util.sh]: For the record, I verified this ID against https://cloud-images.ubuntu.com/locator/

File: cluster/aws/vivid/util.sh [hunk 2]
```diff
@@ -34,6 +34,10 @@ function detect-vivid-image () {
         AWS_IMAGE=ami-ee023e80
         ;;
 
+      ap-northeast-2)
+        AWS_IMAGE=ami-c2df11ac
```
Comment (justinsb) [inline cluster/aws/vivid/util.sh]: For the record, I verified this ID against https://cloud-images.ubuntu.com/locator/

File: cluster/aws/wily/util.sh [hunk 3]
```diff
@@ -33,6 +33,10 @@ function detect-wily-image () {
         AWS_IMAGE=ami-3355505d
         ;;
 
+      ap-northeast-2)
+        AWS_IMAGE=ami-e427e98a
```
Comment (justinsb) [inline cluster/aws/wily/util.sh]: For the record, I verified this ID against https://cloud-images.ubuntu.com/locator/

**Comment by justinsb** [comment]
This looks good, except I don't think you updated your godeps correctly (and it needs a rebase).  I would expect to see an update on Godeps.json, if the Godeps workspace files had changed. (But Godeps never ceases to surprise me, so I could be wrong here!)

File: cluster/aws/vivid/util.sh (hunk 2 above)
Comment (matthewrudy) [inline cluster/aws/vivid/util.sh]: @justinsb great.
when I looked before I couldn't see `ap-northeast-2` on the image finder,
seems I overlooked it right at the bottom of the select box.

**Comment by matthewrudy** [comment]
@justinsb I'll rebase and fix the godeps later today,
thanks.

**Comment by matthewrudy** [comment]
This is good to go @justinsb 

To make the smallest possible change,
I only updated aws-sdk-go to 1.0.8
(the latest version is now 1.1.24)

I issued a separate PR to update to 1.1.24
https://github.com/kubernetes/kubernetes/pull/25411

**Comment by justinsb** [comment]
ok to test

(lgtm though :-) )

**Comment by matthewrudy** [comment]
I rebased, and had to redo the godeps again after they were updated to the latest version.

**Comment by matthewrudy** [comment]
Great.
thanks @justinsb @wojtek-t!

Look forward to deploying to AWS KR!

**Pull request merged**